sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/DefaultSource.scala

```diff
@@ -61,7 +61,9 @@ class DefaultSource extends FileFormat with DataSourceRegister {
     val firstRow = new LineCsvReader(csvOptions).parseLine(firstLine)
 
     val header = if (csvOptions.headerFlag) {
-      firstRow
+      firstRow.zipWithIndex.map { case (value, index) =>
+        if (value == "" || value == null) s"C$index" else value
```

Comment:
I see Spark allows a empty string as a field. So, I wonder if we should rename this with the index and prefix, `C`. Also, I think `""` will throw an NPE whereas empty string without quotes will produce a correct field because the default of `nullValue` is `""`.